python_black/black.py

```diff
@@ -63,8 +63,6 @@ def read_pyproject_toml(
 
     if not config:
         return default_config, None
-    else:
-        config = {k: str(v) if not isinstance(v, (list, dict)) else v for k, v in config.items()}
 
```

Comment:
The function `read_pyproject_toml`  refers to https://github.com/psf/black/blob/0d1b957d400e2884ad312b4c113ee215effb8256/src/black/__init__.py#L104.
Is it better to keep the code on lines 66 and 67?

Reply:
@thep0y hi, this one convert everything to string except list or dict. This lead to stringified booleans e.g. `"True"` instead of `True`. Also I'm not quite sure why we need any conversion additionally to native toml parser.

Reply:
I found this issue when noticed that by default `python-black` plugin sets `skip-string-normalization` and `skip-magic-trailing-comma` to `false` and when I was trying to "override" it there were no effect. After some investigation I noticed that boolean values from `~/.config/black` or project level `pyproject.toml` are not correctly interpreted (as string instead of pure boolean) which leads to inability to set `skip-string-normalization` or `skip-magic-trailing-comma` to `true`.

Reply:
I don’t have time to test, but if you think it’s okay, I will merge this pr when you confirm in your next comment.

Reply:
I'm tested it locally. No issue found so far.

Tested with default config without overrides https://github.com/thep0y/python-black/blob/95d297616132a1cfbbe9f6d76c0bd7e614d7c103/python_black/black.py#L94-L100
```py
    default_config: Dict[str, Any] = {
        "line_length": DEFAULT_LINE_LENGTH,
        "skip_string_normalization": False,
        "skip_magic_trailing_comma": False,
        "experimental_string_processing": False,
        "include": DEFAULT_INCLUDES,
    }
```

And tested it with overrides via `pyproject.toml` and `~/.config/black` (in projects without `pyproject.toml`)
```toml
[tool.black]
skip-string-normalization = true
skip-magic-trailing-comma = true
line-length = 120
```